File: LeetCode/Python/is_subsequence.py
class Solution(object):
    def isSubsequence(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        
        if len(s) > len(t):
            return False
        
        s_index = 0
            
        for c in t:
            if s_index == len(s):
                return True
            elif c == s[s_index]:
                s_index = s_index + 1
            
        return s_index == len(s) # if last character of t was end of subsequence s

File: LeetCode/C++/is_subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        
        int sSize = s.size();
        int tSize = t.size();

        // subsequence cannot have a greater length than original string
        if (sSize > tSize) return false;

        int sPtr = 0;
        int tPtr = -1; // will be incremented first before comparison

        char c = s[sPtr];
        while (sPtr < sSize){
            
            tPtr++;

            if (t[tPtr] == c){
                sPtr++;
                c = s[sPtr];    
            }else if (tPtr == tSize){
                return false;
            }

        }

        return true;

    }
};
